src/DataValues/BoundedQuantityValue.php

```diff
@@ -0,0 +1,381 @@
+	 *        to the transformation callback.
+	 *
+	 * @param mixed ... Any extra parameters will be passed to the $transformation function.
+	 *
+	 * @throws InvalidArgumentException
+	 * @return self
+	 */
+	public function transform( $newUnit, $transformation ) {
+		if ( !is_callable( $transformation ) ) {
+			throw new InvalidArgumentException( '$transformation must be callable.' );
+		}
+
+		if ( !is_string( $newUnit ) ) {
+			throw new InvalidArgumentException( '$newUnit must be a string. Use "1" as the unit for unit-less quantities.' );
+		}
+
+		if ( $newUnit === '' ) {
+			throw new InvalidArgumentException( '$newUnit must not be empty. Use "1" as the unit for unit-less quantities.' );
+		}
+
+		$oldUnit = $this->unit;
+
+		if ( $newUnit === null ) {
+			$newUnit = $oldUnit;
+		}
+
+		// Apply transformation by calling the $transform callback.
+		// The first argument for the callback is the DataValue to transform. In addition,
+		// any extra arguments given for transform() are passed through.
+		$args = func_get_args();
+		array_shift( $args );
+
+		$args[0] = $this->amount;
+		$amount = call_user_func_array( $transformation, $args );
+
+		$args[0] = $this->upperBound;
+		$upperBound = call_user_func_array( $transformation, $args );
+
+		$args[0] = $this->lowerBound;
+		$lowerBound = call_user_func_array( $transformation, $args );
+
+		// use a preliminary QuantityValue to determine the significant number of digits
+		$transformed = new self( $amount, $newUnit, $upperBound, $lowerBound );
+		$roundingExponent = $transformed->getOrderOfUncertainty();
+
+		// apply rounding to the significant digits
+		$math = new DecimalMath();
+
+		$amount = $math->roundToExponent( $amount, $roundingExponent );
+		$upperBound = $math->roundToExponent( $upperBound, $roundingExponent );
+		$lowerBound = $math->roundToExponent( $lowerBound, $roundingExponent );
+
+		return new self( $amount, $newUnit, $upperBound, $lowerBound );
+	}
+
+	public function __toString() {
+		return $this->amount->getValue()
+			. '[' . $this->lowerBound->getValue()
+			. '..' . $this->upperBound->getValue()
+			. ']'
+			. ( $this->unit === '1' ? '' : $this->unit );
+	}
+
+	/**
+	 * @see DataValue::getArrayValue
+	 *
+	 * @since 0.1
+	 *
+	 * @return array
+	 */
+	public function getArrayValue() {
+		return array(
+			'amount' => $this->amount->getArrayValue(),
+			'unit' => $this->unit,
+			'upperBound' => $this->upperBound->getArrayValue(),
+			'lowerBound' => $this->lowerBound->getArrayValue(),
+		);
+	}
+
+	/**
+	 * Constructs a new instance of the DataValue from the provided data.
+	 * This can round-trip with @see getArrayValue
+	 *
+	 * @since 0.1
+	 *
+	 * @param mixed $data
+	 *
+	 * @return self
+	 * @throws IllegalValueException
+	 */
+	public static function newFromArray( $data ) {
+		self::requireArrayFields( $data, array( 'amount', 'unit', 'upperBound', 'lowerBound' ) );
+
+		return new static(
```

Comment:
This could call the parent class when the two fields are missing (for convenience).